src/source-netmap.c

```diff
@@ -385,14 +386,15 @@ static int NetmapOpen(char *ifname, int promisc, NetmapDevice **pdevice, int ver
                            strerror(errno));
                 break;
             }
-            pdev->nif = NETMAP_IF(pdev->mem, nm_req.nr_offset);
-        }
+            pdev->nif = nifp = NETMAP_IF(pdev->mem, nm_req.nr_offset);
+        } else
+		nifp = NETMAP_IF(pdev->mem, nm_req.nr_offset);
```

Comment:
Remove `else` and always compute the nifp before using `NETMAP_RXRING` and `NETMAP_TXRING`.